schema_based_on_diagram.sql

```diff
@@ -0,0 +1,85 @@
+CREATE TABLE "medical_histories"(
+    "id" INTEGER NOT NULL,
+    "admitted_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "patient_id" INTEGER NOT NULL,
+    "status" VARCHAR(255) NOT NULL
+);
+
+ALTER TABLE
+    "medical_histories"
+ADD
+    PRIMARY KEY("id");
+
+CREATE TABLE "invoices"(
+    "id" INTEGER NOT NULL,
+    "total_amount" DECIMAL(8, 2) NOT NULL,
+    "generated_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "payed_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "medical_history_id" INTEGER NOT NULL
+);
+
+ALTER TABLE
+    "invoices"
+ADD
+    PRIMARY KEY("id");
+
+CREATE TABLE "invoice_items"(
+    "id" INTEGER NOT NULL,
+    "unique_price" DECIMAL(8, 2) NOT NULL,
+    "quantity" INTEGER NOT NULL,
+    "total_price" DECIMAL(8, 2) NOT NULL,
+    "invoice_id" INTEGER NOT NULL,
+    "treatment_id" INTEGER NOT NULL
+);
+
+ALTER TABLE
+    "invoice_items"
+ADD
+    PRIMARY KEY("id");
+
+CREATE TABLE "treatments"(
+    "id" INTEGER NOT NULL,
+    "type" VARCHAR(255) NOT NULL,
+    "name" VARCHAR(255) NOT NULL
+);
+
+ALTER TABLE
+    "treatments"
+ADD
+    PRIMARY KEY("id");
+
+CREATE TABLE "patients"(
+    "id" INTEGER NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "date_of_birth" DATE NOT NULL
+);
+
+ALTER TABLE
+    "patients"
+ADD
+    PRIMARY KEY("id");
+
+ALTER TABLE
+    "medical_histories"
+ADD
+    CONSTRAINT "medical_histories_patient_id_foreign" FOREIGN KEY("patient_id") REFERENCES "patients"("id");
+
+ALTER TABLE
+    "invoice_items"
+ADD
+    CONSTRAINT "invoice_items_treatment_id_foreign" FOREIGN KEY("treatment_id") REFERENCES "treatments"("id");
+
+ALTER TABLE
+    "invoices"
+ADD
+    CONSTRAINT "invoices_medical_history_id_foreign" FOREIGN KEY("medical_history_id") REFERENCES "medical_histories"("id");
+
+ALTER TABLE
+    "treatments"
+ADD
+    CONSTRAINT "treatments_id_foreign" FOREIGN KEY("id") REFERENCES "medical_histories"("id");
+
```

Comment:
- [ ] Kindly note that there is a many-to-many relationship between the `medical_histories` table and the `treatments` table. Therefore we should have a join table between the two tables containing the `medical_history_id` as FOREIGN KEYS  referencing the medical_histories(id), and `treatments_id` of treatments as FOREIGN KEYS referencing the treatments(id).